app/models/recommendation.rb

```diff
@@ -15,8 +15,8 @@
 
 class Recommendation < ActiveRecord::Base
   has_and_belongs_to_many :books
-  has_and_belongs_to_many :user_countries, class_name:"Country", foreign_key:'user_country_id'
-  has_and_belongs_to_many :book_countries, class_name:"Country", foreign_key:'book_country_id'
-  has_and_belongs_to_many :user_languages, class_name:"Language", foreign_key:'user_language_id'
-  has_and_belongs_to_many :book_languages, class_name:"Language", foreign_key:'book_language_id'
+  has_and_belongs_to_many :user_countries, class_name:"Country", join_table:"countries_recommendations", association_foreign_key:'user_country_id'
+  has_and_belongs_to_many :book_countries, class_name:"Country", join_table:"countries_recommendations", association_foreign_key:'book_country_id'
+  has_and_belongs_to_many :user_languages, class_name:"Language", join_table:"languages_recommendations", association_foreign_key:'user_language_id'
```

Comment:
Line is too long. [148/80]<br>Space missing after colon.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.<br>Prefer `has_many :through` to `has_and_belongs_to_many`.